pkg/util/certificate/certificate.go

```diff
@@ -0,0 +1,30 @@
+package certificate
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"crypto/x509"
+	"math"
+	"time"
+)
+
+const DefaultMinDurationPercent = 0.20
+
+// lessThanMinimumDuration indicates whether the provided cert has less
+// than the provided minimum percentage of its duration remaining.
+func LessThanMinimumDuration(cert *x509.Certificate, minDurationPercent float64) bool {
+	expiry := cert.NotAfter
+	duration := expiry.Sub(cert.NotBefore)
+	minDuration := time.Duration(float64(duration.Nanoseconds()) * DefaultMinDurationPercent)
+	replacementTime := expiry.Add(-minDuration)
+	return time.Now().After(replacementTime)
+}
+
+func IsCertExpired(cert *x509.Certificate) bool {
+	return (DaysUntilExpiration(cert) <= 0)
```

Comment:
Since you have the cert just return whether we're after the NotAfter time.
```suggestion
	return time.Now().After(cert.NotAfter)
```

Reply:
Done.